File: ScalaLecture/src/main/scala/DSA/BubbleSort.scala
package DSA

object BubbleSort {
  def main(args: Array[String]): Unit = {
    val data: Array[Int] = Array(4,1,2,6,7,5,3)

    for (i <- 0 until data.length - 1; j <- i until data.length - i - 1 ) {
      if (data(j) > data(j + 1)) {
        val temp = data(j)
        data(j) = data(j+1)
        data(j+1) = temp
      }
    }

    println(s"Sorted data: ${data.mkString(", ")}")
  }
}

File: java/Basic/src/DSA/BubbleSort.java
package DSA;

import java.util.StringJoiner;

public class BubbleSort {
    public static void main(String[] args) {
        int[] dataArray = { 7, 2, 5, 1, 4, 3, 6 };

        for (int i = 0; i < dataArray.length - 1; i++) {
            for (int j = 0; j < dataArray.length - 1 - i; j++) {
                if (dataArray[j] > dataArray[j + 1]) {
                    int temp = dataArray[j];
                    dataArray[j] = dataArray[j + 1];
                    dataArray[j + 1] = temp;
                }
            }
        }

        for (int i : dataArray) {
            System.out.printf("%d, ", i);
        }

    }
}
